lib/foreman_maintain/cli/backup_command.rb

```diff
@@ -57,6 +57,7 @@ def common_backup_options
           option '--features', 'FEATURES',
             "#{proxy_name} features to include in the backup. " \
                 'Valid features are tftp, dns, dhcp, openscap, and all.', :multivalued => true
+          option '--wait-for-tasks', :flag, 'Wait for running tasks to complete instead of aborting'
```

Comment:
Right now I've implemented this as a boolean flag, but am wondering if instead we want to give users the ability to configure the timeout. (the current one is 300, so 5 minutes, for both pulp and foreman, so a maximum of 10 minutes waiting)

Reply:
Yes - however we could make it explicit (options to the CLI) or "hidden" (environment variables).

Reply:
I think that is the only one that is left undecided.
I must admit I kinda find env vars bad for discovery, but on the other hand we'd have to add the cli command to update/upgrade too if we'd go with that (to be consistent).
But I also really have no feeling how often people would actually want to override that.

So, maybe, start with env, and then see how many complaints we get?